src/compiler/checker.ts

```diff
@@ -41313,6 +41326,21 @@ namespace ts {
             return false;
         }
 
+        function checkGrammarImportCallArguments(node: ImportCall, nodeArguments: NodeArray<Expression>): boolean {
+            if (moduleKind !== ModuleKind.ESNext) {
+                if (nodeArguments.length !== 1) {
+                    return grammarErrorOnNode(node, Diagnostics.Dynamic_import_must_have_one_specifier_as_an_argument);

```

Comment:
This error is not helpful. Maybe you should add a new error for import assertion in older modules

Reply:
Would you mind give some suggestions?

Reply:
Or, `the second argument of an import call is only supported when the module flag is not available in the current module flag`